assessments/projects/openfga/joint-assessment.md

```diff
@@ -0,0 +1,647 @@
+|MITRE classification|TA0001: Initial Access -> T1078: Valid Accounts<br>TA0003: Persistence -> T1078: Valid Accounts<br>TA0004: Privilege Escalation -> T1078: Valid Accounts|
+|Actors|openfga.server|
+|Suggested Mitigation|Secrets mounted in filesystem can be restricted with permissions, however, may not offer a significantly higher level of security.<br>SPIFFE/ Spire integration may offer a much high level of security.|
+|Impact (High/ Med/ Low)|High|
+|Likelihood (High/ Med/ Low)|Low|
+
+### Setting secure defaults for install
+
+This section has findings related to default install options that can be made more 
+secure. The artifacts analysed in this section include the helm chart for installation 
+and default configuration options.
+
+|Summary|Open playground link - clients can access the playground (enabled by default) without authorization and access openfga models.|
+|--|--|
+|Discovered in self-assessment?|Yes|
+|Weakness|Unauthorized access to openfga data. Ability to both view and manipulate data.|
+|MITRE classification|TA0043: Reconnaissance -> T1595: Active Scanning<br>TA0001: Initial Access -> T1189: Drive-by Compromise|
+|Actors|openfga.server|
+|Suggested Mitigation|Already bring addresed by project team - <a href="https://github.com/openfga/roadmap/issues/7">https://github.com/openfga/roadmap/issues/7</a>.<br>Documentation is clear to disable playground for prod deployments.|
+|Impact (High/ Med/ Low)|High|
+|Likelihood (High/ Med/ Low)|Low|
+
+|Summary|Helm chart runs containers with higher privilege by default.|
+|--|--|
+|Discovered in self-assessment?|No|
+|Weakness|Unauthorized access to openfga data. Ability to both view and manipulate data.|
+|MITRE classification|TA0002: Execution -> T1203: Exploitation for Client Execution<br>TA0003: Persistence -> T1098: Account Manipulation<br>TA0004: Privilege Escalation -> T1548: Abuse Elevation Control Mechanism<br>TA0004: Privilege Escalation -> T1134: Access Token Manipulation<br>TA0004: Privilege Escalation -> T1098: Account Manipulation<br>TA0004: Privilege Escalation -> T1611: Escape to Host<br>TA0005: Defense Evasion -> T1548: Abuse Elevation Control Mechanism<br>TA0005: Defense Evasion -> T1134: Access Token Manipulation|
+|Actors|openfga.server|
+|Suggested Mitigation|Could the defaults for all install scripts be set to run the openfga server with limited permissions?<br>In the case of helm chart, this would achieve: <ol><li>Not running server as root</li> <li>Not allowing privilege escalation</li> <li>Not allowing access to system calls unless required</li> <li>Setting filesystem to readonly </li> <li>Limiting access to mounted filesystems</li></ol><br>This would greatly reduce the attack surface area. |
+|Impact (High/ Med/ Low)|High|
+|Likelihood (High/ Med/ Low)|Low|
+
+|Summary|An external dependency to groundnuty/k8s-wait-for is pinned using tag.|
```

Comment:
Created issue https://github.com/openfga/helm-charts/issues/144

Reply:
Have added a link to the project where all issues are created at the end of the document to refer to next steps from the project.